### PR TITLE
[Osi] Upgrade to v0.108.10

### PR DESCRIPTION
**Pull request by amontoison**
@odow

### CODE REVIEW

File: C/Coin-OR/coin-or-common.jl [hunk 1]
```diff
@@ -42,11 +42,13 @@ Cgl_gitsha = "8952b9e737e434b730fab5967cd28180b43d7234"
...
-Osi_gitsha = "c67528fc8b2a36dc936b285fe34030439b9147c3"
+Osi_version = offset_version(v"0.108.10", v"0.0.0")
+Osi_gitsha = "2997cda8e85ccc6712c4b05404e7aa70500e422f"
 
-CoinUtils_version = offset_version(v"2.11.6", v"0.0.0")
-CoinUtils_gitsha = "26e9639ed9897e13e89169870dbe910296a9783b"
+CoinUtils_upstream_version = v"2.11.11"
+CoinUtils_version = offset_version(CoinUtils_upstream_version, v"0.0.0")
+CoinUtils_gitsha = "15a819b7e8763b2557e4bf440f8cb62ee6734f36"
+CoinUtils_hash = "27da344479f38c82112d738501643dcb229e4ee96a5f87d4f406456bdc1b2cb4"
```
Comment (odow) [inline C/Coin-OR/coin-or-common.jl:51]: You might need to rebase? Why was this needed?

File: C/Coin-OR/Osi/build_tarballs.jl [hunk 2]
```diff
@@ -58,9 +67,9 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("CoinUtils_jll", compat="$(CoinUtils_version)"),
-    Dependency("OpenBLAS32_jll", OpenBLAS32_version),
-    Dependency("CompilerSupportLibraries_jll"),
+    Dependency(PackageSpec(name="CoinUtils_jll", uuid="be027038-0da8-5614-b30d-e42594cb92df"), compat="$(CoinUtils_version)"),
```
Comment (odow) [inline C/Coin-OR/Osi/build_tarballs.jl:70]:
```suggestion
    Dependency(PackageSpec(name="CoinUtils_jll", uuid="be027038-0da8-5614-b30d-e42594cb92df"), compat="=$(CoinUtils_version)"),
```

Comment (odow) [reply C/Coin-OR/Osi/build_tarballs.jl:70]: We need to ensure strict compat `=` for all COIN-OR stuff going forward.

**Comment by odow** [comment]
Looks like the failure was just that the registry hadn't propagated yet.

File: C/Coin-OR/Osi/build_tarballs.jl (hunk 2 above)
Comment (amontoison) [reply C/Coin-OR/Osi/build_tarballs.jl]: Ok, let's do like Ipopt.
In theory we don't need the strict compat `=` but as we observed in the past, updating one dependency could break all COIN-OR packages.

**Review by odow — APPROVED** [review]
LGTM, assuming CI passes

File: C/Coin-OR/Osi/build_tarballs.jl [hunk 3]
```diff
@@ -74,5 +83,5 @@ build_tarballs(
     products,
     dependencies;
     preferred_gcc_version = gcc_version,
-    julia_compat = "1.6",
+    julia_compat = "1.9",
```
Comment (odow) [inline C/Coin-OR/Osi/build_tarballs.jl:86]: Do we need a preferred llvm version here so that it matches the CoinUtils one?

Comment (odow) [inline C/Coin-OR/Osi/build_tarballs.jl:86]:
```suggestion
    preferred_llvm_version = llvm_version,
    julia_compat = "1.9",
```

Comment (amontoison) [reply C/Coin-OR/Osi/build_tarballs.jl:87]: I think that we need is `clang_use_lld = false`.
The issue is the new linker on Mac that doesn't support some options.
Mosè added this option after my recipe for `CoinUtils`.

But using an older version of `llvm` should also ensure that.

**Comment by amontoison** [comment]
The next one is `Clp` (#8066).
